### PR TITLE
Enrichment

### PR DESCRIPTION
**Pull request by wxli0**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,6 +56,8 @@
     "file-saver": "^1.3.3",
     "fready": "^1.0.0",
     "got": "^7.1.0",
+    "gulp": "^3.9.1",
```
Comment (d2fong) [inline package.json]: Do you really need these dependencies?

File: src/client/features/gene-query/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//const func = require('../../../server/gene-query/validate');
+//const validate = func.validate;
+//const proceed = func.proceed;
+
+class GeneQuery extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+    this.handleChange = this.handleChange.bind(this);
```
Comment (d2fong) [inline src/client/features/gene-query/index.js]: delete this line

File: src/client/features/gene-query/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    this.handleSubmit = this.handleSubmit.bind(this);
```
Comment (d2fong) [inline src/client/features/gene-query/index.js]: delete this line

File: src/client/features/gene-query/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  handleSubmit(event) {
+    const result = ServerAPI.geneQuery(this.state.value);
+    alert('A name was submitted: ' + result);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
```
Comment (d2fong) [inline src/client/features/gene-query/index.js]: You should be using hyperscript

Comment (d2fong) [reply src/client/features/gene-query/index.js]: See any of the other react components for reference.

File: src/server/enrichment/enrichment.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
+const request = require('request');
+const Promise = require('promise-simple');
+const csv = require('csvtojson');
+const fs = require('fs');
+const _ = require('lodash');
+const enrichment = function (query, user_setting) {
+  var default_setting = {
```
Comment (d2fong) [inline src/server/enrichment/enrichment.js]: use let or const don't use var

File: src/server/routes/rest.js [hunk 6]
```diff
@@ -57,11 +60,26 @@ router.get('/get-graph-and-layout', function (req, res) {
   });
 });
 
+// expose a rest endpoint for validator
 router.get('/gene-query/', (req, res) => {
-  const genes = req.query.genes;
+  const genes = req.query.gene;
+  console.log("endpoint genes are: "+genes);
```
Comment (d2fong) [inline src/server/routes/rest.js]: dont put console.log statements into committed code.

**Comment by wxli0** [comment]
Can I make those changes on another branch? The most uptodate branch is 'validate_info'.

**Comment by d2fong** [comment]
ok

**Comment by wxli0** [comment]
Committed on validator_info.

**Comment by d2fong** [comment]
closing old prs

**Pull request closed without merging**